src/genobs/include/genobs/observable.hpp

```diff
@@ -0,0 +1,41 @@
+#ifndef INCLUDE_OBSERVABLES_OBSERVABLE_HPP
+#define INCLUDE_OBSERVABLES_OBSERVABLE_HPP
+
+#include "algorithms.hpp"
+#include "properties.hpp"
+
+namespace GenObs {
+/**
+ * @brief Meta-Observable that returns the product of two
+ *        other observables.
+ *
+ * The operand observables are stored by privately deriving
+ * from them to get the empty case optimization if they do
+ * not have state.
+ *
+ * @tparam Left left operand of the product.
+ * @tparam Right right operand of the product.
+ */
+template <class Left, class Right> struct Product : Left, Right {
+  Product(Left left = {}, Right right = {})
+      : Left(std::move(left)), Right(std::move(right)) {}
+
+  template <class Particle, class Traits = default_traits<Particle>>
+  decltype(auto) operator()(Particle const &p,
+                            Traits const &traits = {}) const {
+    return Left::template operator()<Particle, Traits>(p, traits) *
+           Right::template operator()<Particle, Traits>(p, traits);
+  }
+};
+
+using Momentum = Product<Mass, Velocity>;
+using AverageMomentum = Average<Momentum>;
+using CenterOfMassPosition = WeightedAverage<Position, Mass>;
+using CenterOfMassVelocity = WeightedAverage<Velocity, Mass>;
+using TotalForce = Sum<Force>;
+using Positions = Map<Position>;
+using Velocities = Map<Velocity>;
+using Forces = Map<Force>;
```

Comment:
Should we write down a Doxygen page to give an overview of how to construct new observables in the core and how to integrate them with accumulators, correlators, and the ScriptInterface? What I have in mind is something similar to  http://espressomd.org/html/dox/script_interface.html

Reply:
the new observable library does not interfere with the ScriptInterface, we should probably document the ```ParticleObservable``` class. The naming is probably also not the best yet.

Reply:
In principle you should replace `PidObservables`, then you could reuse the name.

Reply:
I think before we can properly document this we have to clarify some concepts here with regards to thee `PidObservable`, to me the use of `PidObservable` for the bonds seems
slightly abusive, because it does not generalize to other particle filters (e.g. dihedrals between subsequent particlles of type 5 does not make sense). But this is out of scope. It would be nice to have a dockstring on `ParticleObservable` thou.